```
aniket@DESKTOP-074O80J:/mnt/d/napari/napari$ uv run --no-sync mypy --config-file pyproject.toml --pretty --show-error-codes src/napari 2>&1 | grep -A 3 "qt_scrollbar"
src/napari/_qt/widgets/qt_scrollbar.py:5: error: Module "qtpy.QtCore" has no
attribute "Qt"  [attr-defined]
    from qtpy.QtCore import Qt
    ^~~~~~~~~~~~~~~~~~~~~~~~~~
src/napari/_qt/widgets/qt_scrollbar.py:34: error: Item "None" of "Any | None"
has no attribute "position"  [union-attr]
                event.position().toPoint()
                ^~~~~~~~~~~~~~
src/napari/_qt/widgets/qt_scrollbar.py:36: error: Item "None" of "Any | None"
has no attribute "pos"  [union-attr]
                else event.pos()
                     ^~~~~~~~~
src/napari/_qt/widgets/qt_scrollbar.py:75: error: Item "None" of "Any | None"
has no attribute "buttons"  [union-attr]
            if event.buttons() & Qt.MouseButton.LeftButton:
               ^~~~~~~~~~~~~
src/napari/_qt/widgets/qt_scrollbar.py:81: error: Item "None" of "Any | None"
has no attribute "button"  [union-attr]
            if event.button() == Qt.MouseButton.LeftButton:
               ^~~~~~~~~~~~
aniket@DESKTOP-074O80J:/mnt/d/napari/napari$
```

ahh 😿 just figured out i think i was getting different mypy errors locally due PyQt version

--- a/src/napari/_qt/widgets/qt_scrollbar.py
+++ b/src/napari/_qt/widgets/qt_scrollbar.py
@@ -2,7 +2,7 @@ from __future__ import annotations
 
 from typing import TYPE_CHECKING
 
-from qtpy.QtCore import Qt  # type: ignore[attr-defined]
+from qtpy.QtCore import Qt
 from qtpy.QtWidgets import QScrollBar, QStyle, QStyleOptionSlider
 
 if TYPE_CHECKING:
@@ -24,7 +24,9 @@ class ModifiedScrollBar(QScrollBar):
     fully to the clicked position.
     """
 
-    def _move_to_mouse_position(self, event: QMouseEvent) -> None:
+    def _move_to_mouse_position(self, event: QMouseEvent | None) -> None:
+        if event is None:
+            return
         opt = QStyleOptionSlider()
         self.initStyleOption(opt)
 
@@ -35,16 +37,19 @@ class ModifiedScrollBar(QScrollBar):
             if hasattr(event, 'position')
             else event.pos()
         )
-        control = self.style().hitTestComplexControl(
+        style = self.style()
+        if style is None:
+            return
+        control = style.hitTestComplexControl(
             CC.CC_ScrollBar, opt, point, self
         )
         if control not in {SC.SC_ScrollBarAddPage, SC.SC_ScrollBarSubPage}:
             return
         # scroll here
-        gr = self.style().subControlRect(
+        gr = style.subControlRect(
             CC.CC_ScrollBar, opt, SC.SC_ScrollBarGroove, self
         )
-        sr = self.style().subControlRect(
+        sr = style.subControlRect(
             CC.CC_ScrollBar, opt, SC.SC_ScrollBarSlider, self
         )
         if self.orientation() == Qt.Orientation.Horizontal:
@@ -69,13 +74,17 @@ class ModifiedScrollBar(QScrollBar):
             )
         )
 
-    def mouseMoveEvent(self, event: QMouseEvent) -> None:
+    def mouseMoveEvent(self, event: QMouseEvent | None) -> None:
+        if event is None:
+            return None
         if event.buttons() & Qt.MouseButton.LeftButton:
             # dragging with the mouse button down should move the slider
             self._move_to_mouse_position(event)
         return super().mouseMoveEvent(event)
 
-    def mousePressEvent(self, event: QMouseEvent) -> None:
+    def mousePressEvent(self, event: QMouseEvent | None) -> None:
+        if event is None:
+            return None
         if event.button() == Qt.MouseButton.LeftButton:
             # clicking the mouse button should move slider to the clicked point
             self._move_to_mouse_position(event)

